base/src/main/java/com/smartdevicelink/managers/screen/BaseTextAndGraphicManager.java

```diff
@@ -741,53 +742,64 @@ private boolean sdlArtworkNeedsUpload(SdlArtwork artwork){
-		if (defaultMainWindowCapability == null || defaultMainWindowCapability.getImageTypeSupported() == null || defaultMainWindowCapability.getImageTypeSupported().size() > 0) {
-			if (currentScreenData.getGraphic() == null) {
-				return primaryGraphic != null;
-			} else {
-				return currentScreenData != null
-						&& (primaryGraphic != null && !CompareUtils.areStringsEqual(currentScreenData.getGraphic().getValue(), primaryGraphic.getName(), true, true));
-			}
-		}
-		return false;
+		boolean templateSupportsPrimaryArtwork = templateSupportsImageField(ImageFieldName.graphic);
+
+		String currentScreenDataPrimaryGraphicName = (currentScreenData != null && currentScreenData.getGraphic() != null) ? currentScreenData.getGraphic().getValue() : null;
+		String primaryGraphicName = primaryGraphic != null ? primaryGraphic.getName() : null;
+		return templateSupportsPrimaryArtwork
+				&& !CompareUtils.areStringsEqual(currentScreenDataPrimaryGraphicName, primaryGraphicName, true, true)
+				&& primaryGraphic != null;
 	}
 
+	/**
+	 * Check to see if secondaryGraphic should be updated
+	 * @return true if secondaryGraphic should be updated, false if not
+	 */
 	private boolean shouldUpdateSecondaryImage() {
-		// Cannot detect if there is a secondary image, so we'll just try to detect if there's a primary image and allow it if there is.
-		if (defaultMainWindowCapability == null || defaultMainWindowCapability.getImageTypeSupported() == null || defaultMainWindowCapability.getImageTypeSupported().size() > 0) {
-			if (currentScreenData.getGraphic() == null) {
-				return secondaryGraphic != null;
-			} else {
-				return currentScreenData != null
-						&& (secondaryGraphic != null && !CompareUtils.areStringsEqual(currentScreenData.getGraphic().getValue(), secondaryGraphic.getName(), true, true));
-			}
-		}
-		return false;
-	}
+		boolean templateSupportsSecondaryArtwork = (templateSupportsImageField(ImageFieldName.graphic) || templateSupportsImageField(ImageFieldName.secondaryGraphic));
 
-	int getNumberOfLines() {
+		String currentScreenDataSecondaryGraphicName = (currentScreenData != null && currentScreenData.getSecondaryGraphic() != null) ? currentScreenData.getSecondaryGraphic().getValue() : null;
+		String secondaryGraphicName = secondaryGraphic != null ? secondaryGraphic.getName() : null;
+		return templateSupportsSecondaryArtwork
+				&& !CompareUtils.areStringsEqual(currentScreenDataSecondaryGraphicName, secondaryGraphicName, true, true)
+				&& secondaryGraphic != null;
+	}
 
-		if (defaultMainWindowCapability == null){
-			return 4;
-		}
+	/**
+	 * Check to see if template supports the specified image field
+	 * @return true if image field is supported, false if not
+	 */
+	private boolean templateSupportsImageField(ImageFieldName name) {
+		return (defaultMainWindowCapability == null || defaultMainWindowCapability.getImageFields() == null) || ManagerUtility.WindowCapabilityUtility.hasImageFieldOfName(defaultMainWindowCapability, name);
```

Comment:
I don't believe this logic is correct. Based on the RPC spec, the `imageField` array is not mandatory. Therefore if it is not included, we should infer that it is not supported, right? @joeljfischer can you confirm?